lib/source-code/token-store/utils.js

```diff
@@ -30,9 +30,20 @@ function getStartLocation(token) {
  * @returns {number} The found index or `tokens.length`.
  */
 exports.search = function search(tokens, location) {
-    const index = tokens.findIndex(el => location <= getStartLocation(el));
+    for (let minIndex = 0, maxIndex = tokens.length - 1; minIndex <= maxIndex;) {
+        const index = Math.trunc((minIndex + maxIndex) / 2);
+        const token = tokens[index];
 
-    return index === -1 ? tokens.length : index;
+        if (location <= getStartLocation(token)) {
```

Comment:
If `getStartLocation(token)` is inlined as `token.range[0]`, performance seems to improve.